woocommerce/woocommerce-ios
# Product description/sharing AI: Add feedback banner

Part of #10101 


## Description

This PR adds a new banner at the bottom of product description AI and product sharing AI sheets to collect feedback about AI-generated content. Changes include:
- Added a reusable banner to collect feedback named `FeedbackView` following design: NMncL2ViLjomfLhuyEhVM1-fi.
- Added `FeedbackView` to both product description and sharing AI sheets.
- Added analytics to track feedback.

## Testing instructions

Product Sharing:
- Log in to a public WPCom site.
- Select a published product or create one.
- Tap to share the product. On the product sharing AI sheet, tap Write with AI.
- When a message is generated, notice a new banner for feedback.
- Tap the button to upvote or downvote on the feedback banner. Notic that the button state is updated, and the banner is dismissed afterward.
- Notice an event is logged in Xcode console: `🔵 Tracked product_ai_feedback, properties: [AnyHashable("is_useful"): true, AnyHashable("blog_id"): <id>, AnyHashable("source"): "product_sharing_message", AnyHashable("is_wpcom_store"): true]`.
- Tap Regenerate, and a new feedback banner should be displayed.

Product Description:
- Log in to a WPCom site.
- Select an existing product or create one.
- Tap Write with AI on the product description row to show the AI sheet for the product description. Tap Write with AI button on the sheet.
- When a message is generated, notice a new banner for feedback.
- Tap the button to upvote or downvote on the feedback banner. Notice that the button state is updated, and the banner is dismissed afterward.
- Notice an event is logged in Xcode console: `🔵 Tracked product_ai_feedback, properties: [AnyHashable("blog_id"): <id>, AnyHashable("is_wpcom_store"): true, AnyHashable("is_useful"): false, AnyHashable("source"): "product_description"]`.
- Tap Regenerate, and a new feedback banner should be displayed.

Known issue: language detection seems to fail again for product description. I created a separate issue here: https://github.com/woocommerce/woocommerce-ios/issues/10103.

Unit tests will be updated in a subsequent PR: #10104 

## Screenshots


Product Sharing:
https://github.com/woocommerce/woocommerce-ios/assets/5533851/78e74728-8940-4d9e-b094-06f11c6c4cb6


Product Description:

https://github.com/woocommerce/woocommerce-ios/assets/5533851/e6e33f28-7b3a-4e00-85c6-f6a5723df7dd



---
